spec/support/fetchable_behavior.rb

```diff
@@ -136,3 +173,41 @@
     end
   end
 end
+
+shared_examples_for 'a record with an indexable url' do
+  describe 'validations' do
+    context 'when the url extension is blacklisted' do
+      let(:movie_url) { "http://www.nps.gov/some.mov" }
+      let(:record) { described_class.new(valid_attributes.merge(url: movie_url)) }
+
+      it "is not valid" do
+        expect(record).not_to be_valid
+        expect(record.errors.full_messages.first).to match(/extension is not one we index/i)
+      end
+    end
+  end
+end
+
+shared_examples_for 'a record that belongs to a searchgov_domain' do
+  describe 'associations' do
+    it { is_expected.to belong_to(:searchgov_domain) }
+
+    context 'on creation' do
+      context 'when the domain already exists' do
+        let!(:existing_domain) { SearchgovDomain.create!(domain: 'existing.gov') }
+        let!(:sitemap) { described_class.create!(url: 'https://existing.gov/foo') }
+
+        it 'sets the searchgov domain' do
+          expect(sitemap.searchgov_domain).to eq(existing_domain)
+        end
+      end
+
+      context 'when the domain has not been created yet' do
+        it 'creates the domain' do
+          expect{ described_class.create!(url: 'https://brandnewdomain.gov/loop') }.
```

Comment:
I also noticed that in the transfer of this spec from `searchgov_url_spec.rb` to `fetchable_behavior.rb`, the url changed from `https://brandnewdomain.gov/foo` to `https://brandnewdomain.gov/loop`. I will fix that; I'm guessing that was a text-editor-induced typo?

Reply:
I think so!